pkg/tsdb/stackdriver/stackdriver.go

```diff
@@ -351,40 +351,60 @@ func (e *StackdriverExecutor) unmarshalResponse(res *http.Response) (Stackdriver
 }
 
 func (e *StackdriverExecutor) parseResponse(queryRes *tsdb.QueryResult, data StackdriverResponse, query *StackdriverQuery) error {
-	metricLabels := make(map[string][]string)
-	resourceLabels := make(map[string][]string)
-	var resourceTypes []string
-
-	for _, series := range data.TimeSeries {
-		if !containsLabel(resourceTypes, series.Resource.Type) {
-			resourceTypes = append(resourceTypes, series.Resource.Type)
-		}
-	}
+	labels := make(map[string][]string)
+	labels["resource.type"] = make([]string, 0)
 
 	for _, series := range data.TimeSeries {
 		points := make([]tsdb.TimePoint, 0)
+		seriesLabels := make(map[string]string)
 
 		defaultMetricName := series.Metric.Type
-		if len(resourceTypes) > 1 {
-			defaultMetricName += " " + series.Resource.Type
+
+		seriesLabels["resource.type"] = series.Resource.Type
+		if !containsLabel(labels["resource.type"], series.Resource.Type) {
+			labels["resource.type"] = append(labels["resource.type"], series.Resource.Type)
 		}
 
 		for key, value := range series.Metric.Labels {
-			if !containsLabel(metricLabels[key], value) {
-				metricLabels[key] = append(metricLabels[key], value)
+			if _, exist := labels["metric.label."+key]; !exist {
+				labels["metric.label."+key] = make([]string, 0)
```

Comment:
These `!exist` checks and assignments appear redundant. The following `containsLabel()` returns `false` either way, and `append()` will allocate as-needed.

For example, https://play.golang.org/p/qjovfSISn2U

Reply:
Yes, of course. Thanks for feedback!